### ISSUE TITLE
hyperkube kubelet: KillPodSandboxError ... networkPlugin cni failed to teardown pod ... network: neither iptables nor ip6tables usable

### ISSUE THREAD

**Issue by waynr** [kubernetes/kubernetes#92272]
**What happened**:

* I attempted to bump the 1.16, 1.17, and 1.18 versions in my team's kubernetes image repo to 1.16.11, 1.17.7, and 1.18.4, respectively.
* The images built successfully but none of the e2e tests we run, including the kubernetes conformance e2e tests, ever actually ran correctly without a clear indication as to why -- ginkgo was just mysteriously succeeding without any tests passed, skipped, or failed. this prompted me to take a closer look a 1.18.4 cluster.
* I expected corednes pods to be in "Running" status but instead found:
```
kube-system   coredns-6b6854dcbf-tqxgn          0/1     Completed   0          5h24m
kube-system   coredns-6b6854dcbf-wztzn          0/1     Completed   0          5h24m
```
* ...and the following event on one of those pods:
```
  Normal  SandboxChanged  98s (x627 over 136m)  kubelet, test-1-18-4-0b8b9360-3fhas  Pod sandbox changed, it will be killed and re-created.
```
* This even prompted me to log on to a worker node and notice repeated messages like the following in the kubelet logs:
```
Jun 18 21:41:53 test-1-18-4-06729635-3f4er docker[1211]: E0618 21:41:53.316852    1291 pod_workers.go:191] Error syncing pod f364dc8e-6e7d-4869-976a-678a3ce913ac ("coredns-6b6854dcbf-2t9cv_kube-system(f364dc8e-6e7d-4869-976a-678a3ce913ac)"), skipping: failed to "KillPodSandbox" for "f364dc8e-6e7d-4869-976a-678a3ce913ac" with KillPodSandboxError: "rpc error: code = Unknown desc = networkPlugin cni failed to teardown pod \"coredns-6b6854dcbf-2t9cv_kube-system\" network: neither iptables nor ip6tables usable"
```

This led me to grep through the kubernetes repo at v1.18.4 for the message `network: neither iptables nor ip6tables usable` but I didn't find anything. So I did the same thing in the containernetworking plugins repo and found the following references:

* https://github.com/containernetworking/plugins/blob/ad10b6fa91aacd720f1f9ab94341a97a82a24965/plugins/meta/portmap/portmap.go#L120
* https://github.com/containernetworking/plugins/blob/ad10b6fa91aacd720f1f9ab94341a97a82a24965/plugins/meta/portmap/portmap.go#L384

It's kind of hard to tell which of the possible `maybeGetIptables` function failures occurred because it doesn't lift the error into the calling context but this led me to believe that there is something different about the iptables installed in the hyperkube container that we are are using to run our kubelets. And I did find a potentially significant difference at least between 1.18.3 and 1.18.4 (i haven't compared the patch level versions on 1.16 and 1.17 yet):

```
zsh/4 10048 % docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.3 iptables --version
iptables v1.6.0
```

```
zsh/4 10047 % docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.4 iptables --version
iptables v1.8.2 (nf_tables)
```

I'm not familiar enough with iptables to say with any kind of certainty that this is really the root cause of the problem, but if not then it's a pretty good red herring.

**What you expected to happen**:

I was hoping that `1.16.11`, `1.17.7`, and `1.18.4` would "just work" similar to the previous patch level versions.

**How to reproduce it (as minimally and precisely as possible)**:

I'm so far from really understanding what's going on at pretty much all times that it's actually kind of depressing but without just dumping a bunch of configuration files that I don't think would be very helpful, here is what I think may be happening:

We have `--network-plugin=cni` passed to kubelet in our systemd unit file, and some part of kubelet is somehow delegating networking stuff to the CNI binaries that appear to be copied into the hyperkube image:
```
zsh/4 10049 % docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.4 ls /opt/cni/bin
bandwidth  bridge  dhcp  firewall  flannel  host-device  host-local  ipvlan  loopback  macvlan  portmap  ptp  sbr  static  tuning  vlan
```

So I'm guessing that `portmap` here corresponds to the code I linked to above in the cni plugins repo.

**Anything else we need to know?**:

I considered the possibility of *not* filing this bug and instead chiming in on #92242 or #92250 but I thought it would be better to post a new issue rather than risk the information I've gathed being lost in the noise of comments.

**Environment**: 
- Kubernetes version (use `kubectl version`):
1.16.11, 1.17.7, and 1.18.4
- Cloud provider or hardware configuration:
Digitalocean
- OS (e.g: `cat /etc/os-release`):
```
zsh/4 10051 % docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.3 cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

```
zsh/4 10054 [2] % docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.4 cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
- Kernel (e.g. `uname -a`):
```
Linux test-1-18-4-06729635-3f4er 4.19.0-0.bpo.6-amd64 #1 SMP Debian 4.19.67-2+deb10u2~bpo9+1 (2019-11-12) x86_64 GNU/Linux
```

- Install tools:
hyperkube

- Network plugin and version (if this is a network-related bug):
happens on both v0.7.4 and v0.8.6 of https://github.com/containernetworking/plugins

(we volume mount a directory containing the cni plugin binaries from our worker node host into the kubelet hyperkube containers so it appears that we can end up using a different CNI version than upstream -- I don't know why we do this, maybe in the past hyperkube didn't include the binaries or maybe we needed a specific version of CNI temporarily and just never went back to using the bundled binaries; regardless, because I've tried both versions I assume this isn't the root cause of the problem I'm seeing)

**Comment by waynr** [comment]
So after thinking about this a little more, I think the problem for me might be the use of debian 9 as a base image rather than debian 10, since according to https://wiki.debian.org/nftables:

> Starting with Debian Buster, nf_tables is the default backend when using iptables, by means of the iptables-nft layer (i.e, using iptables syntax with the nf_tables kernel subsystem). This also affects ip6tables, arptables and ebtables. 

So my options are probably:

* configure my images starting with 1.16.11, 1.17.7, and 1.18.4 to use the nftables backend for iptables since in debian 9 the default seems to be the legacy backend (whatever that means)
* try to upgrade to debian 10 as the base image starting with 1.16.11, 1.17.7, and 1.18.4

In other words, what I've reported probably isn't a bug in kubernetes or hyperkube (although it might be worth mentioning the iptables version change in the release notes). Although arguably upgrading to the newer version of iptables without setting the update-alternatives to iptables-legacy is a backwards-incompatible change...

**Comment by waynr** [comment]
Okay I think I've gotten past the previous error but now I am running into a missing package:

incriminating kubelet log message:
```
Jun 19 02:11:52 validate-debian-10-default-pool-3fibj docker[1160]: gops: accept tcp 127.0.0.1:34147: use of closed network connectionW0619 02:11:52.051584    1184 docker_sandbox.go:394] failed to read pod IP from plugin/docker: networkPlugin cni failed on the status hook f
or pod "coredns-858c66b796-hg2sq_kube-system": unexpected command output nsenter: failed to execute ip: No such file or directory
```

iproute2 installed in v1.18.3:
```
zsh/4 10017  (git)-[master]-% docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.3 apt search iproute2
Sorting... Done
Full Text Search... Done
iproute2/now 4.9.0-1+deb9u1 amd64 [installed,local]
  networking and traffic control tools
```

iproute2 not installed in v1.18.4:
```
zsh/4 10024  (git)-[master]-% docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.4 /bin/bash -c 'apt update && apt search iproute2'
Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:2 http://deb.debian.org/debian buster InRelease [121 kB]
Get:3 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
Get:4 http://security.debian.org/debian-security buster/updates/main amd64 Packages [203 kB]
Get:5 http://deb.debian.org/debian buster/main amd64 Packages [7905 kB]
Get:6 http://deb.debian.org/debian buster-updates/main amd64 Packages [7868 B]
Fetched 8355 kB in 6s (1367 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
7 packages can be upgraded. Run 'apt list --upgradable' to see them.
Sorting... Done
Full Text Search... Done
iproute2/stable 4.20.0-2 amd64
  networking and traffic control tools

iproute2-doc/stable 4.20.0-2 all
  networking and traffic control tools - documentation
```

**Comment by justaugustus** [comment]
/assign
/priority critical-urgent
/area release-eng

I'm attempting a fix in https://github.com/kubernetes/kubernetes/pull/92354.

**Comment by justaugustus** [comment]
cc: @kubernetes/sig-network-bugs @kubernetes/release-engineering

**Comment by athenabot** [comment]
/triage unresolved

Comment `/remove-triage unresolved` when the issue is assessed and confirmed.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by justaugustus** [comment]
We've published [v1.18.5-rc.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.18.5-rc.1), [v1.17.8-rc.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.17.8-rc.1), and [v1.16.12-rc.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.16.12-rc.1), which include new hyperkube images.

Can you test these and let us know if this resolves your issue?
We're holding off on any patch releases until we get feedback here, so please let report back when you can.

cc: @kubernetes/release-engineering

**Comment by waynr** [comment]
@justaugustus after working around the way our mount paths horribly break the hyperkube image's (perfectly reasonable) use of `alternatives` (we inexplicably decided at one point in the past to mount `/etc:/etc` and `/var:/var` into our kubelet container and I had no success in breaking these mounts down into smaller, actually necessary mounts that don't break `alternatives`) I once again encountered the other problem I reported in this issue:

```
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]: I0626 14:01:58.639877    2707 kuberuntime_manager.go:735] Created PodSandbox "f1815b86b79b26c115c5ba44a27acb5faed0699f9bf44986b806cea3785eb70c" for pod "coredns-6b6854dcbf-mvbbf_kube-system(a966fd45-d549-417c-a80b-daa4dcc7c81e)"
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]: W0626 14:01:58.646743    2707 docker_sandbox.go:400] failed to read pod IP from plugin/docker: networkPlugin cni failed on the status hook for pod "coredns-6b6854dcbf-mvbbf_kube-system": unexpected command output nsenter: failed to execute ip: No such file or directory
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]:  with error: exit status 127
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]: I0626 14:01:58.648588    2707 kuberuntime_manager.go:754] Determined the ip [] for pod "coredns-6b6854dcbf-mvbbf_kube-system(a966fd45-d549-417c-a80b-daa4dcc7c81e)" after sandbox changed
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]: I0626 14:01:58.649494    2707 kuberuntime_manager.go:853] checking backoff for container "coredns" in pod "coredns-6b6854dcbf-mvbbf_kube-system(a966fd45-d549-417c-a80b-daa4dcc7c81e)"
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]: I0626 14:01:58.649816    2707 kuberuntime_manager.go:863] back-off 20s restarting failed container=coredns pod=coredns-6b6854dcbf-mvbbf_kube-system(a966fd45-d549-417c-a80b-daa4dcc7c81e)
```

More specifically:

```
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]: W0626 14:01:58.646743    2707 docker_sandbox.go:400] failed to read pod IP from plugin/docker: networkPlugin cni failed on the status hook for pod "coredns-6b6854dcbf-mvbbf_kube-system": unexpected command output nsenter: failed to execute ip: No such file or directory
Jun 26 14:01:58 meow-default-pool-3yho8 docker[2683]:  with error: exit status 127
```

```
zsh/3 10027 [1] % docker run --rm -it --entrypoint "" gcr.io/google-containers/hyperkube:v1.18.5-rc.1 /bin/bash
root@b87fc85c4074:/# which ip
root@b87fc85c4074:/# apt update && apt search iproute2
Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:2 http://deb.debian.org/debian buster InRelease [121 kB]
Get:3 http://security.debian.org/debian-security buster/updates/main amd64 Packages [204 kB]
Get:4 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
Get:5 http://deb.debian.org/debian buster-backports InRelease [46.7 kB]
Get:6 http://deb.debian.org/debian buster/main amd64 Packages [7905 kB]
Get:7 http://deb.debian.org/debian buster-updates/main amd64 Packages [7868 B]
Get:8 http://deb.debian.org/debian buster-backports/main amd64 Packages [328 kB]
Fetched 8730 kB in 6s (1531 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
5 packages can be upgraded. Run 'apt list --upgradable' to see them.
Sorting... Done
Full Text Search... Done
iproute2/stable 4.20.0-2 amd64
  networking and traffic control tools

iproute2-doc/stable 4.20.0-2 all
  networking and traffic control tools - documentation
```

This is the problem my PR, https://github.com/kubernetes/kubernetes/pull/92281, attempted to fix.

**Comment by waynr** [comment]
Oh and just to clarify, attaching to the kubelet container and installing iproute2 appears to have fixed the worker node where this was happening.

**Comment by justaugustus** [comment]
@waynr -- Opened https://github.com/kubernetes/kubernetes/pull/92581 for this.
ref: https://github.com/kubernetes/kubernetes/pull/92281#issuecomment-650717496

**Comment by athenabot** [comment]
@justaugustus
If this issue has been triaged, please comment `/remove-triage unresolved`.

If you aren't able to handle this issue, consider unassigning yourself and/or adding the `help-wanted` label.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by liggitt** [comment]
@justaugustus is this resolved now?

**Comment by dominicgunn** [comment]
This is still broken for us in v1.16.13. Infact it's broken for us in any version _after_ v1.16.10. We're currently pinned, because otherwise all manner of different behaviour occurs whenever try and turn on the calico CNI. For completeness sake, we're also using hyperkube, and roll exactly the same configuration to our clusters, the **only** different is the kubelet version (*.10, vs *.13).

**Comment by waynr** [comment]
@dominicgunn for what it's worth there was a significant amount of workaround we had to do to accommodate the new Debian 10 base image in Hyperkube even after the upstream fixes -- for example, we had to write our own iptables wrapper script that exec's the correct ipables binary directly rather than going through the alternatives system since we mount top-level directories like `/etc` into the kubelet hyperkube container (awful, i know).

But after these additional hacks the latest releases do appear to work for us.

**Comment by dominicgunn** [comment]
Thanks for sharing @waynr, we could probably also get it working if we stuck a bunch of workarounds in place too, but really I (or others) should be able to safely update the patch version, this is still completely breaking change.

**Comment by dominicgunn** [comment]
Is this still being actively investigated @justaugustus? This is still broken.

**Comment by justaugustus** [comment]
@dominicgunn -- reports like this are exactly why hyperkube is deprecated. See this recent thread for further context: https://groups.google.com/d/msg/kubernetes-sig-release/2Eyoxk0muGI/KVnc12cdAgAJ

We have no tests running against hyperkube to validate functionality and no one who has stepped to own this over multiple cycles.
Right now, we're prioritizing security updates over anything else, which resulted in a break in functionality for you.

So to echo my plea [here](https://groups.google.com/d/msg/kubernetes-announce/-jsxn9kCTo8/qCWvh2YyAAAJ):
> As a general note, hyperkube is deprecated and will NOT be present in the v1.19.0 release or any releases > v1.19.0.
> **_Please, if you're using hyperkube for older releases (v1.18.z --> v1.16.z), make plans to migrate away from it as a deployment strategy.
We have extremely limited capabilities to test hyperkube and support for it should be considered to be "best-effort"._**

---

Despite that, we want to try to do what we can to get you back in a working state, but I have neither the bandwidth or context to help right now.

Marko, given I think you likely have the most hyperkube experience on the team, can you work with the reporters to try and move towards a resolution here?

/unassign
/assign @xmudrii

**Comment by dominicgunn** [comment]
Wow, I hadn't even realized hyperkube was deprecated, I'm so out of the loop, thanks for sharing that.

**Comment by xmudrii** [comment]
Hey folks! I've been busy for the past few days, but I'll be looking into this tomorrow.

@waynr If possible, can you please share some more details about how you got this working, how the wrapper script looks like, and to which iptables binary you point?

**Comment by dominicgunn** [comment]
Sorry to be naggy, did you manage to get anywhere with this @xmudrii?

**Comment by xmudrii** [comment]
@dominicgunn I've looked into it a little bit, but I've been busy for the past few days. I'll do my best to take a deeper look tonight or tomorrow.

**Comment by xmudrii** [comment]
@waynr @dominicgunn I've tried to reproduce this issue with the latest patch releases, but I was not able to do so. Instead, I've run into some other issues and not so usable clusters, but I believe that it's not related to the Kubelet issue.

My setup was the following:

* One control plane and worker node on AWS
* Debian 10 (Buster) running on both instances
* Cluster provisioned with kubeadm using Hyperkube images for all components
  * The kubeadm and kubectl binaries were installed manually, as installing deb packages would install kubelet as well
* Hyperkube Kubelet running in a Docker container using the `k8s.gcr.io/hyperkube` image
* kubernetes-cni v0.8.6
* Tested with Calico, Flannel, Canal, Weave CNI plugins
* Tested with 1.18.8 and 1.16.14

What have I encountered:

* Both nodes in the cluster were ready
```
NAME                                           STATUS   ROLES    AGE   VERSION
ip-172-31-108-221.eu-west-3.compute.internal   Ready    master   16m   v1.18.8
ip-172-31-108-95.eu-west-3.compute.internal    Ready    <none>   10m   v1.18.8
```
* All pods in the cluster were Running, including CoreDNS pods (there were no Completed, CrashLoopBackoff, or Pending pods)
```
NAMESPACE     NAME                                                                   READY   STATUS    RESTARTS   AGE
kube-system   calico-kube-controllers-578894d4cd-52685                               1/1     Running   0          12m
kube-system   canal-4qhwl                                                            2/2     Running   0          12m
kube-system   canal-8976x                                                            2/2     Running   0          11m
kube-system   coredns-66bff467f8-lgm89                                               0/1     Running   0          11m
kube-system   coredns-66bff467f8-tgrx2                                               0/1     Running   0          11m
kube-system   etcd-ip-172-31-108-221.eu-west-3.compute.internal                      1/1     Running   0          16m
kube-system   kube-apiserver-ip-172-31-108-221.eu-west-3.compute.internal            1/1     Running   0          15m
kube-system   kube-controller-manager-ip-172-31-108-221.eu-west-3.compute.internal   1/1     Running   0          15m
kube-system   kube-proxy-8zv8f                                                       1/1     Running   0          11m
kube-system   kube-proxy-hfk6s                                                       1/1     Running   0          16m
kube-system   kube-scheduler-ip-172-31-108-221.eu-west-3.compute.internal            1/1     Running   0          15m
```
* The cluster networking was entirely broken. Pod-to-pod or service communication was not working at all.
  * I assume that this has nothing to do with Kubelet at all.
  * Kubelet shows no errors this time. Using a command from one of the comments above (`journalctl -fu kubelet | grep -C 10 ' ip '`) gives no result on both nodes.
  * I tried recreating clusters with different CNIs, but it didn't help at all.
  * CoreDNS was showing the following logs:
```
[INFO] plugin/ready: Still waiting on: "kubernetes"
.:53
[INFO] plugin/reload: Running configuration MD5 = 4e235fcc3696966e76816bcd9034ebc7
CoreDNS-1.6.7
linux/amd64, go1.13.6, da7f65b
[ERROR] plugin/errors: 2 3729317466437900974.1938216730541960538. HINFO: read udp 10.244.1.3:46738->172.31.0.2:53: i/o timeout
[ERROR] plugin/errors: 2 3729317466437900974.1938216730541960538. HINFO: read udp 10.244.1.3:37075->172.31.0.2:53: i/o timeout
[INFO] plugin/ready: Still waiting on: "kubernetes"
[ERROR] plugin/errors: 2 3729317466437900974.1938216730541960538. HINFO: read udp 10.244.1.3:35106->172.31.0.2:53: i/o timeout
[ERROR] plugin/errors: 2 3729317466437900974.1938216730541960538. HINFO: read udp 10.244.1.3:39475->172.31.0.2:53: i/o timeout
[ERROR] plugin/errors: 2 3729317466437900974.1938216730541960538. HINFO: read udp 10.244.1.3:39655->172.31.0.2:53: i/o timeout
[ERROR] plugin/errors: 2 3729317466437900974.1938216730541960538. HINFO: read udp 10.244.1.3:46617->172.31.0.2:53: i/o timeout
```
  * I've tried switching to legacy iptables as described here (https://github.com/coredns/coredns/issues/2693#issuecomment-543208513), but that didn't help.
  * I've tried to `nslookup` any service from any pod in the cluster, but that didn't work:
```
If you don't see a command prompt, try pressing enter.
nslookup: can't resolve 'kubernetes.default'
pod "busybox" deleted
pod default/busybox terminated (Error)
```
  * kube-proxy doesn't show any errors. It uses the `iptables` proxier.

It would be great if @waynr and @dominicgunn can test the latest patch releases and report does it work for them. If Kubelet is still a problem, information about the setup, how they provisioned the cluster, and how they run Kubelet, would be useful/needed to proceed. If Kubelet is not a problem anymore, I'd propose that we close this issue in favor of a new one.

I think that it might be that kube-proxy is misbehaving, but I had no time to check the iptables rules and verify is everything correct.

cc @justaugustus

**Comment by waynr** [comment]
The latest patch releases are working for my team both with and without hyperkube (but we have eliminated hyperkube from our builds anyway). Because of the super janky way we were using hyperkube to run kubelet, we actually had to create our own iptables wrapper script that we would bind mount into the container since a couple of our volume mounts were totally breaking the alternatives system (we ran debian 9 as our node OS and newer hyperkube images were incompatibly upgraded to debian 10 on patch level versions, which prompted me to open this issue in the first place).

If I remember correctly our wrapper script basically looks something like:

```
#!/usr/bin/env bash

exec /usr/sbin/iptables-legacy $@
```

Again, this worked around the alternatives database (used by the hyperkube iptables wrapper script) broken by the way the docker run command for our kubelet container volume-mounted the top-level `/etc` and `/var` directories which caused the debian 10 hyperkube container was to see debian 9 alternatives configuration & symlinks. This caused all iptables commands run in the hyperkube containers in our test clusters to break.

However, we have entirely moved away from the use of hyperkube at this point. If your worker/master nodes are running kubelet in a hyperkube-based container and you're running on an OS that has a [supported kubelet binary installation method](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl) anyway, you're wasting your time with hyperkube. Note that you shouldn't need to also use kubeadm as described in linked installation instructions; if you already have static manifests for kube-apiserver, kube-controller-manager, and kube-scheduler then you just need to replace kuberkube in those manifests with the component-specific images that can be found [here](https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/). 

The Kubernetes release team has made it clear enough that there is nothing more they are going to do to support hyperkube and after my experience wrangling a really awful hyperkube-based kubelet into submission I see why they wouldn't want to. I would close this issue since the release team did actually address enough of the problems I originally described that I was able to proceed, but it looks like enough people are still having trouble and appear to think that trouble is related to the original problems that I'll leave issue closure for someone else with more authority.

**Comment by aojea** [comment]
>  I would close this issue since the release team did actually address enough of the problems I originally described that I was able to proceed, but it looks like enough people are still having trouble and appear to think that trouble is related to the original problems that I'll leave issue closure for someone else with more authority.

@LappleApple should we close? seems the user that reported it is ok with closing it and the issue is stale right now
